.github/workflows/deployment.yaml

```
@@ -0,0 +1,83 @@
+name: Deployment
+
+on:
+  push:
+    branches:
+      - master
+      - LISK-1400-Setup-Dshackle
+
+env:
+  ECS_TASK_DEFINITION: task-definition.json
+
+jobs:
+  docker:
+    name: Deploy
+    permissions:
+      id-token: write
+      contents: read
+    runs-on: ubuntu-latest
+    environment: ${{ github.ref_name }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+          submodules: recursive
+
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: "zulu"
+          java-version: 21
+
+      - name: Setup gradle
+        uses: gradle/gradle-build-action@v2
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@0e613a0980cbf65ed5b322eb7a1e075d28913a83
```

Let's switch to using the [v4 tag](https://github.com/aws-actions/configure-aws-credentials/releases/tag/v4).

```suggestion
        uses: aws-actions/configure-aws-credentials@v4
```